src/SmartFormat/Extensions/ListFormatter.cs

```diff
@@ -241,15 +241,15 @@ public bool TryEvaluateFormat(IFormattingInfo formattingInfo)
                 }
                 else if (CollectionIndex < items.Count - 1)
                 {
-                    spacerFormattingInfo.Write(spacer);
+                    spacerFormattingInfo.FormatAsChild(spacer, item);
```

Comment:
`item` is just an element from the list, that spacers will hardly use?

What if the argument to `FormatChild`was the data context of the root parent like so:
```CSharp
// Note:
// Give spacers the data context of the root parent.
// formattingInfo.CurrentValue from the argument to
// TryEvaluateFormat(IFormattingInfo formattingInfo) only contains the list elements.
var rootParent = (formattingInfo as Core.Formatting.FormattingInfo);
do
{
    rootParent = rootParent?.Parent;
} while (rootParent?.Parent != null);

var rootParentValue = rootParent?.CurrentValue;

foreach (var item in items)
{
...
}
```
This also had the advantage, that it's no necessary to refer to the argument index, which currently is a **must**.
Another test on top of
```Csharp
Assert.AreEqual("John, Mary nor Amy", smart.Format("{0.Names:list:{}|, | {0.Not:nor|or} }", new { Names = names, Not = true }));
```
could then (maybe more intuitively) read
```Csharp
Assert.AreEqual("John, Mary nor Amy", smart.Format("{Names:list:{}|, | {Not:nor|or} }", new { Names = names, Not = true }));
```

Reply:
Yeah that makes sense